Parser/asdl_c.py

```diff
@@ -637,7 +643,13 @@ def visitField(self, field, name, sum=None, prod=None, depth=0):
             self.emit("if (tmp == NULL || tmp == Py_None) {", depth)
             self.emit("Py_CLEAR(tmp);", depth+1)
             if self.isNumeric(field):
-                self.emit("%s = 0;" % field.name, depth+1)
+                if field.name in self.attribute_special_defaults:
+                    self.emit(
+                        "%s = %s;" % (field.name, self.attribute_special_defaults[field.name]),
```

Comment:
Can f-string be used here for a little shorter string?

```suggestion
                        f'{field.name} = {self.attribute_special_defaults[field.name]};'
```